src/pip/_internal/commands/index.py

```diff
@@ -134,6 +137,21 @@ def get_available_package_versions(self, options: Values, args: List[Any]) -> No
             formatted_versions = [str(ver) for ver in sorted(versions, reverse=True)]
             latest = formatted_versions[0]
```

Comment:
Is this the way that pip has been using to determine the the latest version of a package?
This does not always sound right, given the version string can contain non-numerical characters.
Do we have a standard on the package version naming convention so that the latest package version always gets ranked first after sorting?
If we don't have such a standard, should we rank them by their release time?

Reply:
The version objects are orderable, so calling sorted on them is doing the right thing.

Reply:
`versions` is a set of `Version` objects. They sort according to the standard. r. See the documentation for the `packaging` project, and the [version specifiers standard](https://packaging.python.org/en/latest/specifications/version-specifiers/).

Reply:
@pfmoore I see. So it's the user's responsibility to follow the standard and ensure the latest package version always gets ranked first.

Reply:
If by "latest" you mean "most recently released", then yes. Release date is irrelevant, only version number matters.